protocol/messenger_test.go

```diff
@@ -2540,3 +2540,237 @@ func (s *MessengerSuite) TestSendMessageMention() {
 	s.Require().NoError(err)
 	s.Require().Equal("Alice talk to bobby", response.Notifications()[0].Message)
 }
+
+func (s *MessengerSuite) TestMarkMessageAsUnread() {
+
+	s.T().Run("Test marking a message as unread when message list contains 1 message", func(t *testing.T) {
+		chat := CreatePublicChat("test-chat-1", s.m.transport)
+
+		chat.UnviewedMessagesCount = 2
+		chat.UnviewedMentionsCount = 2
+		chat.Highlight = true
+
+		err := s.m.SaveChat(chat)
+		s.Require().NoError(err)
+
+		inputMessage1 := buildTestMessage(*chat)
+		inputMessage1.ID = "1"
+		inputMessage1.Seen = true
+		inputMessage1.Mentioned = true
+
+		err = s.m.SaveMessages([]*common.Message{inputMessage1})
+		s.Require().NoError(err)
+
+		_, err = s.m.MarkAllRead(context.Background(), chat.ID)
+		s.Require().NoError(err)
+
+		chats := s.m.Chats()
+
+		for idx := range chats {
```

Comment:
This function can be used to avoid using a loop:
https://github.com/status-im/status-go/blob/3ab1afaae8c01c3228a8349fb606b767e7d697dc/protocol/messenger_chats.go#L135-L139

Reply:
sure, I can update this. Thanks